lib/auth/auth_with_roles_test.go

```diff
@@ -8398,3 +8398,185 @@ func TestIsMFARequired_AdminAction(t *testing.T) {
+			token := resetToken.GetName()
+
+			registerChal, err := srv.AuthServer.CreateRegisterChallenge(ctx, &proto.CreateRegisterChallengeRequest{
+				TokenID:     token,
+				DeviceType:  proto.DeviceType_DEVICE_TYPE_WEBAUTHN,
+				DeviceUsage: proto.DeviceUsage_DEVICE_USAGE_PASSWORDLESS,
+			})
+			require.NoError(t, err)
+
+			_, registerSolved, err := NewTestDeviceFromChallenge(registerChal, WithPasswordless())
+			require.NoError(t, err)
+
+			setupAuthContext, err := srv.Authorizer.Authorize(authz.ContextWithUser(ctx, TestAdmin().I))
+			require.NoError(t, err)
+
+			// actual authentication method change via ServerWithRoles
+			serverWithRoles := &ServerWithRoles{
+				authServer: srv.AuthServer,
+				alog:       srv.AuditLog,
+				context:    *setupAuthContext,
+			}
+
+			serverWithRoles.ChangeUserAuthentication(ctx, &proto.ChangeUserAuthenticationRequest{
+				TokenID:                token,
+				NewMFARegisterResponse: registerSolved,
+			})
+
+			authPreferences, err := serverWithRoles.GetAuthPreference(ctx)
+			require.NoError(t, err)
+
+			// assert that the auth preference matches the expected
+			require.Equal(t, tc.expectedDefaultConnector, authPreferences.GetConnectorName())
+		})
+	}
+}
+
+func TestHasOneNonPresetUser(t *testing.T) {
+	tt := []struct {
+		name            string
+		users           []types.User
+		existingPresets []types.User
+		expected        bool
+	}{
+		{
+			name: "One user should return true",
+			users: []types.User{&types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user1",
+				},
+			}},
+			existingPresets: getPresetUsers(),
+			expected:        true,
+		},
+		{
+			name: "Two users should return false",
+			users: []types.User{&types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user1",
+				},
+			}, &types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user2",
+				},
+			}},
+			existingPresets: getPresetUsers(),
+			expected:        false,
+		},
+		{
+			name: "Manually removed presets, returns false with three users",
+			users: []types.User{&types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user1",
+				},
+			}, &types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user2",
+				},
+			}, &types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user3",
+				},
+			}},
+			// no existing presets to simulate them being removed after the cluster started.
+			existingPresets: []types.User{},
+			expected:        false,
+		},
+		{
+			name: "Manually removed presets, returns true with one users",
+			users: []types.User{&types.UserV2{
+				Metadata: types.Metadata{
+					Name: "user1",
+				},
+			}},
+			// no existing presets to simulate them being removed after the cluster started.
+			existingPresets: []types.User{},
+			expected:        true,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			// append preset to users list
+			users := append(tc.users, tc.existingPresets...)
+			actual := hasOneNonPresetUser(users)
```

Comment:
I don't think we need to test the private method directly, this should already be covered by TestCloudDefaultPasswordless.